tools/analyzer_plugin/lib/src/diagnostic/missing_required_prop.dart

```diff
@@ -52,82 +56,115 @@ class MissingRequiredPropDiagnostic extends ComponentUsageDiagnosticContributor
+        if (requiredFieldsByName.containsKey(propField.name)) {
           // Short-circuit if we've already identified this field as required.
           // There might be duplicates if props are overridden, and there will
           // definitely be duplicates in the builder-generated code.
           continue;
         }
 
-        if (field.metadata.any((annotation) {
-          // Common case, might be good to short circuit here for perf
-          if (annotation.isOverride) return false;
-
-          final value = annotation.computeConstantValue();
-          if (value == null) return false;
-
-          final type = value.type;
-          if (type == null) return false;
-
-          final typeLibrary = type.element?.library;
-          if (typeLibrary == null || typeLibrary.name != 'over_react.component_declaration.annotations') {
-            return false;
-          }
-
-          // ignore: unnecessary_parenthesis
-          final accessorClass = (_cachedAccessorClass ??= typeLibrary.getClass('Accessor')!);
-          if (!result.typeSystem.isAssignableTo(type, accessorClass.thisType)) {
-            return false;
-          }
-
-          // This is null when isRequired does not have a valid value
-          // (e.g., as the user is typing it in)
-          return value.getField('isRequired')!.toBoolValue() ?? false;
-        })) {
-          requiredFields.add(field);
+        if (isRequiredProp(propField)) {
+          requiredFieldsByName[propField.name] = propField;
         }
       }
     }
+    final debugHelper = AnalyzerDebugHelper(result, collector, enabled: _debugCommentPattern.hasMatch(result.content));
+    debugHelper.logWithLocation('Required props: $requiredFieldsByName', result.locationFor(usage.builder));
+
+    final missingRequiredFieldNames = requiredFieldsByName.keys.toSet()
+      ..removeAll(usage.cascadedProps.where((prop) => !prop.isPrefixed).map((prop) => prop.name.name));
 
-    final missingRequiredFieldNames = requiredFields.map((f) => f.name).toSet()
-      ..removeAll(usage.cascadedProps.map((prop) => prop.name.name));
 
     for (final name in missingRequiredFieldNames) {
-      collector.addError(
+      await collector.addErrorWithFix(
         code,
         result.locationFor(usage.builder),
         errorMessageArgs: [name],
-        // todo fix to add prop
+        fixKind: fixKind,
+        fixMessageArgs: [name],
+        computeFix: () => buildFileEdit(result, (builder) {
+          addProp(usage, builder, result.content, result.lineInfo, name: name, buildValueEdit: (builder) {
+            builder.selectHere();
+          });
+        }),
       );
     }
   }
 }
+
+extension on InterfaceElement {
+  Iterable<InterfaceElement> get thisAndSupertypes sync* {
+    yield this;
+    for (final s in allSupertypes) {
+      yield s.element;
+    }
+  }
+
+  Iterable<FieldElement> get allProps sync* {
+    for (final c in thisAndSupertypes) {
+      // FIXME handle legacy boilerplate prop mixins?
```

Comment:
Yeah it may not be worth it, based on how many usages are still out there, and how much more effort it would take to get working.

I think by the "?" in my comment, I meant to imply "if it's worth it" 😄